github_visualization/github_stats.py

```diff
@@ -0,0 +1,289 @@
+                                "avatar_url": "https://avatars.githubusercontent.com/u/134276?v=3",
+                                "html_url": "https://github.com/Garyfallidis",
+                                "total_commits": 1389,
+                                "total_additions": 116712,
+                                "total_deletions": 70340,
+                                "weekly_commits": [
+                                          {
+                                            "w": "1367712000",
+                                            "a": 6898,
+                                            "d": 77,
+                                            "c": 10
+                                          },
+                                        ]
+                                },
+                            ]
+              'cumulative_contributors': [
+                                (
+                                    1367712000, // timestamp
+                                    20 // total number of contributions
+                                       // upto this time
+                                ),
+                            ]
+            }
+
+        '''
+        contributor_stats = {}
+        contributor_stats["contributors"] = []
+
+        url = self.baseurl + "/stats/contributors"
+        try:
+            response = requests.get(url)
+            contributor_stats["response_code"] = response.status_code
+
+            # check response code
+            if response.status_code != 200:
+                return contributor_stats
+
+            # parse response
+            r_json = response.json()
+            contributor_stats["total_contributors"] = len(r_json)
+
+            grand_total_commits = 0
+            for contributor in r_json:
+                contributor_dict = {}
+                contributor_dict["user_name"] = contributor["author"]["login"]
+                contributor_dict["avatar_url"] = contributor[
+                    "author"]["avatar_url"]
+
+                contributor_dict["html_url"] = contributor[
+                    "author"]["html_url"]
+
+                contributor_dict["total_commits"] = contributor["total"]
+                grand_total_commits += contributor["total"]
+
+                total_additions, total_deletions = self.get_total_contributions(
+                    contributor["weeks"])
+
+                contributor_dict["total_additions"] = total_additions
+                contributor_dict["total_deletions"] = total_deletions
+                contributor_dict["weekly_commits"] = contributor["weeks"]
+                contributor_stats["contributors"].insert(0, contributor_dict)
+
+            contributor_stats["total_commits"] = grand_total_commits
+
+            cumulative_contributors = self.get_cumulative_contributors(r_json)
+            contributor_stats[
+                "cumulative_contributors"] = cumulative_contributors
```

Comment:
Ok, I saw your reply, but it doesn't show up in github. Anyway, makes sense.

2016-08-15 9:16 GMT-04:00 Bishakh Ghosh <notifications@github.com>:

> In github_visualization/github_stats.py
> <https://github.com/nipy/dipy_web/pull/15#discussion_r74759230>:
>
> > +                contributor_dict["total_commits"] = contributor["total"]
> > +                grand_total_commits += contributor["total"]
> > +
> > +                total_additions, total_deletions = self.get_total_contributions(
> > +                    contributor["weeks"])
> > +
> > +                contributor_dict["total_additions"] = total_additions
> > +                contributor_dict["total_deletions"] = total_deletions
> > +                contributor_dict["weekly_commits"] = contributor["weeks"]
> > +                contributor_stats["contributors"].insert(0, contributor_dict)
> > +
> > +            contributor_stats["total_commits"] = grand_total_commits
> > +
> > +            cumulative_contributors = self.get_cumulative_contributors(r_json)
> > +            contributor_stats[
> > +                "cumulative_contributors"] = cumulative_contributors
>
> @jchoude <https://github.com/jchoude> here is the cumulative_contributors
> actually. Do you mean something different?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nipy/dipy_web/pull/15/files/544d752148f59db2105e126695f7f6f2b9c43455#r74759230>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACDSeBtsdgnwCuSyDDw6Lxmy6v7ydHcCks5qgGaYgaJpZM4JeWOf>
> .
>